my_enumerable.rb

```diff
@@ -0,0 +1,210 @@
+
+# rubocop:disable all
+module Enumerable
+  def my_each
+    for item in self
+      if block_given?
+        yield(item)
+      else
+        item
+      end
+    end
+  end
+
+  def my_each_with_index
+    index=0
+    for item in self
+      if block_given?
+      yield(item, index)
+      else
+        item
+      end
+      index+=1
+    end
+  end
```

Comment:
- [ ] my_each_with_index
returns an Enumerator if no block is given(just like the orginal `each_with_index`):
`p [1, 2, 3].my_each_with_index` returns `#<Enumerator>`